### ISSUE TITLE
Travis -- issues with extended delays, and travis-ci.org is closing down

### ISSUE THREAD

**Issue by MysterAitch** [javaparser/javaparser#2919]
Looks like Travis is being a pain again.

It seems like it is primarily a travis problem, unrelated to the github api limit problems faced earlier this year (#2582).

A tldr (short summary) is at the end :+1: 

## The "pain point"

In the Travis community forums, and elsewhere online, there are a large number of reports of travis ci builds being queued for extended periods of time (multiple hours) and manual intervention being taken after 16hrs.

Here we've been seeing anywhere between 30mins and 2hrs.

With a workflow of "create PR" -> "wait for travis build" -> "merge pr" (which triggers a travis build" -> "update all other PRs to merge in this new change" -> "wait for that travis build to complete" -> "merge the second pr".

Even at 30mins per build, that's potentially 90mins to merge two PRs...

## Secondary issues

In exploring this, I have since discovered that travis-ci.org is closing down. It is the migration from accounts on .org to .com that is leading to these delays.

> **We are in the process of migrating workers from travis-ci.org to travis-ci.com in preparation to fully shut down .org (or rather, make it read-only) on December, 31st.**
> 
> Unfortunately, this might happen as a result — if we see a job stuck in a queue for more than 16 hours, we will review this on a case by case basis and possibly cancel the job to ensure resources are not being reduced across the platform.
> 
> **Here’s a FAQ to clear things up: https://docs.travis-ci.com/user/migrate/open-source-repository-migration#frequently-asked-questions .**
> 
> **You may want to consider migrating across to .com sooner rather than later**, or consider building at a quieter time of the day if remaining on .org for a while longer.

https://travis-ci.community/t/builds-hang-in-queued-state/10250/6

Here is a visual of the backlog:

https://www.traviscistatus.com/#week

![image](https://user-images.githubusercontent.com/1537214/99131991-a420ad00-260c-11eb-8e97-adcbbc19f00d.png)

## Migration from travis-ci.org to travis-ci.com

It looks pretty seamless to do the actual transition itself, but it has some "gotchas".

The most substantial concern is that community / open source projects now have an allocation of "tokens" to spend on builds. 

It is unclear whether the allocation of tokens is a one-off or recurring. 

There are several forum posts querying this, with no or limited formal response. The confusion arises from some documentation referring to the free plan being labelled as a "trial" and a blog post saying that trial plans expire either when you use the token allocation or after twelve months.

https://travis-ci.community/t/org-com-migration-unexpectedly-comes-with-a-plan-change-for-oss-what-exactly-is-the-new-deal/10567

![image](https://user-images.githubusercontent.com/1537214/99132408-0201c480-260e-11eb-8543-1f6189c6d52b.png)

![image](https://user-images.githubusercontent.com/1537214/99132402-f8785c80-260d-11eb-9551-b05ceb1137ac.png)

Even if it is recurring, I'm not convinced that the default amount is enough for the number of builds we run each month. One workaround could be to only run travis when commits/merges to master are made, or similar. 

> https://docs.travis-ci.com/user/billing-faq#what-if-i-am-building-open-source
> 
> **What if I am building open source? #**
> 
> Each of the Travis CI Plans can have an amount of special OSS credits per month assigned to run builds only on public repositories. To find out more about it please contact the Travis CI support team.
> 

Other projects have reached out to travis to ask for additional credits, but there is mixed feedback about this and there are several reports of projects not even getting a reply yet.

Bottom line is that the migration to .com (from .org) is currently unclear.

## tldr / What next?

... All this is to say that it is, perhaps, time to move away from travis completely. At the very least, until the dust settles.

- Travis is actively hindering development by introducing substantial delays.
- Travis is redundant given that we have appveyor running in addition to a full build matrix running using github actions (ubuntu/mac/windows, on JDK8-15)
- Travis is setup as a "required" build status for PRs to merge. Increasingly we're having to bypass the restriction and merging anyway (confidence given by appveyor and github actions passing, where the alternative is merging one PR per couple of hours or however long travis takes at that particular point in time).
- Travis-ci.org is closing down in December 2020 anyway.

Steps to take if we do disable travis:

- [x] Remove travis.yml (or move it to an archive folder?)
- [ ] Replace the coveralls.io code coverage webhook (see also #2906)
- [ ] Replace the gitter.im webhook
- [x] Remove travis as a required build step for pull requests (replace it with appveyor? or nothing?)
- [ ] Consider migrating some of the above webhooks to appveyor, from travis?
- [x] Edit the branch protection rules for master

Anything else?

**Comment by MysterAitch** [comment]
https://ropensci.org/technotes/2020/11/19/moving-away-travis/


> Sadly, good times came to an end. In 2019 the company was sold to a private equity firm, and soon after the acquisition, a large portion of the senior engineer staff was layed off. Under the new management, open-source users started suffering from significant outages and backlogs, while being pushed towards the new enterprise product travis-ci.com (the original Travis service was hosted on travis-ci.org).
>
> The big blow came earlier this month, when Travis announced a new pricing model which no longer has a generous free tier for open-source projects, and also fully shutting down the old travis-ci.org product by December 31. It is still unclear what exactly the new product will look like, perhaps it can still be useful, but with the direction the company is heading, we recommend exploring other options.
